src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java

```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.jdbc.dao.DriverDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class DriverDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        String insertQuery = "INSERT INTO drivers (name, driver_license) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement =
+                        connection.prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1, driver.getName());
+            insertStatement.setString(2, driver.getLicenseNumber());
+            insertStatement.executeUpdate();
+            ResultSet generatedKeys = insertStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                driver.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert driver: "
+                    + driver + " to DataBase ", e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String getQuery = "SELECT * FROM drivers WHERE is_deleted = FALSE AND id = (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(getQuery)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            while (resultSet.next()) {
+                return Optional.of(getDriverFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get driver by id " + id
+                    + " from DataBase ", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        List<Driver> allDrivers = new ArrayList<>();
+        String getAllQuery = "SELECT * FROM drivers WHERE is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                allDrivers.add(getDriverFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from DataBase ", e);
+        }
+        return allDrivers;
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String updateQuery = "UPDATE drivers SET name = (?), driver_license = (?) WHERE "
```

Comment:
Have some mentor told you to do wrap everything with ()?

Reply:
I think question mark feels lonely

Reply:
the same issue with semicolons